code-challenges.test.js

```diff
@@ -44,13 +74,32 @@ var fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
 var fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
 // Expected output: [-823, 7, 23]
 
+describe("when 'onlyOdd' is called", () => {
+  let onlyOddOutput1 = [-9, 7, 9, 199]
+  let onlyOddOutput2 = [-823, 7, 23]
+  it("returns only odd numbers in array from least to greatest", () => {
+    expect(onlyOdd(fullArr1)).toEqual(onlyOddOutput1)
+    expect(onlyOdd(fullArr2)).toEqual(onlyOddOutput2)
 
-
+  })
+})
 
 // b) Create the function that makes the test pass.
-
-
-
+//create a function that takes an array of many data types
+//filter through array to look for typeof to determine only numbers && check numbers to see if they are odd by using %2
+//sort numbers using sort method
+
+const onlyOdd = (array) => {
+  let numSearch = array.filter(function(value){
+    return (typeof value === "number") && (value % 2 !== 0)
+      })
+      function sortArray(a, b) {
```

Comment:
using a named function works perfectly fine, but it's generally for when we know we'll be using the function in multiple places. Because this sortArray logic is only used once it can be expressed using an arrow function `(a, b) => a - b`